### DISCUSSION TITLE
Fix map issues with librequake multiplayer

### DISCUSSION THREAD

**Discussion by pnahratow** [MissLav/LibreQuake#158] [General]
- Librequake e1m1 is different from Quake e1m1. This is a problem when joining multiplayer games.
- The sourceports that autodownload missing maps will not do so for the stock maps dm1, dm2, etc. So you cannot join games with quake players.
- The map menu when starting a multiplayer server show only original quake maps and not the librequake ones. This is hardcoded in `menu.c`

It would be awesome to be able to just play multiplayer with librequake. Lets fix these issues.
First we need ideas though.

**Comment by MotoLegacy** [comment]
@lavenderdotpet can we enable Discussions for this repository and move this to a discussion? We shouldn't treat this as an Action Item yet, so moving it away from Issues would be great :)

**Comment by lavenderdotpet** [comment]
> @lavenderdotpet can we enable Discussions for this repository and move this to a discussion? We shouldn't treat this as an Action Item yet, so moving it away from Issues would be great :)

we have discussions https://github.com/lavenderdotpet/LibreQuake/discussions

**Comment by MotoLegacy** [comment]
Can you convert it to a discussion then, as per the second part of the question? Thanks

**Comment by Zungrysoft** [comment]
There really isn't a solution for being unable to join matches on original Quake maps. That's a copyright issue. If a server wants to host LibreQuake players, they will need to avoid any original Quake maps. Those are the breaks, unfortunately.

Also, FTEQW's deathmatch start menu lets you type in whichever map filename you want instead of making you pick from a list.

**Comment by Zungrysoft** [comment]
There is a separate but related issue with map collisions. Every time we make a new release, we recompile the map bsp files from source, which means the maps will be ever-so-slightly different from the ones in the previous release, causing an error upon joining the server. Additionally, if a LibreQuake server is hosting one of our singleplayer levels for deathmatch, original Quake players won't be able to join since their e1m1 is different from the server's.

The only solution I can come up with for that is to implement a feature in FTEQW (our recommended engine) that allows you to temporarily download a map even if you already have another map of the same name and load that one instead.
